kernel/term_typing.ml

```diff
@@ -186,7 +192,8 @@ let rec unzip ctx j =
       unzip ctx { j with uj_val = mkApp (mkLambda (n,ty,j.uj_val),arg) }
 
 let hcons_j j =
-  { uj_val = hcons_constr j.uj_val; uj_type = hcons_constr j.uj_type} 
+  (** Do not hashcons type: it is not used by the callers of this function. *)
```

Comment:
It seems that at the beginning, it was passed to some function but that code disappeared in between. I'm going to remove it I think.